src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java

```diff
@@ -28,13 +28,14 @@
     void createBuildPlanForExercise(ProgrammingExercise exercise, String planKey, String repositoryName, String testRepositoryName);
 
     /**
-     * Copy the base build plan for the given user on the CI system.
+     * Clones an existing build plan. Illegal characters in the plan key, or name will be replaced.
      *
-     * @param templateBuildPlanId unique identifier for build plan on CI system
-     * @param username            username of user for whom to copy build plan
-     * @return unique identifier of the copied build plan
+     * @param templatePlan The key of the template plan.
+     * @param planKey The wanted key of the new plan
+     * @param planName The wanted name of the new plan
+     * @return The key of the new build plan
      */
-    String copyBuildPlan(String templateBuildPlanId, String username);
+    String clonePlan(String templatePlan, String planKey, String planName);
```

Comment:
change signature to String copyPlan(String sourceProjectName, String sourceBuildPlanName, String targetProjectName, String targetBuildPlanName)

improve java doc

Reply:
Refactored it